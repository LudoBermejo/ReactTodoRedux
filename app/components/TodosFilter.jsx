import React from 'react';

const TodosFilter = (props) => {
  let searchText;
  let checkCompleted;

  const handleSearch = () => {
    const search = {
      searchText: searchText.value.toLowerCase(),
      filterByComplete: checkCompleted.checked
    }
    props.onSearch(search);
  };

  return (
    <div className="container__header" >
      <div>
        <input
          type="search"
          ref={input => searchText = input}
          placeholder="Search in todos"
          onChange={handleSearch}
        />
      </div>
      <label>
        <input
          type="checkbox"
          ref={input => checkCompleted = input}
          onChange={handleSearch}
        /> Show completed todos
      </label>
    </div>
  );
};

TodosFilter.propTypes = {
  onSearch: React.PropTypes.func.isRequired
};

export default TodosFilter;
