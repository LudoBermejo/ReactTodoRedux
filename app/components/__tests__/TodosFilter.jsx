import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import TodosFilter from 'TodosFilter';

describe('TodosFilter ', () => {
  it('renders properly', () => {
    const mock = jest.fn();
    const todosFilter = renderer.create(<TodosFilter onSearch={mock} onFilterByCompleted={mock} />).toJSON();
    expect(todosFilter).toMatchSnapshot();
  });

  describe('works as expected', () => {
    it('call onSearch if input has changed', () => {
      const mock1 = jest.fn();
      const todosFilter = mount(<TodosFilter onSearch={mock1} />);
      const first = todosFilter.find('input[type="search"]');
      first.node.value = 'fff';
      first.simulate('change', first);
      expect(mock1).toHaveBeenCalledWith({ filterByComplete: false, searchText: 'fff' });
    });

    it('call onSearch if checkbox has changed', () => {
      const mock1 = jest.fn();
      const todosFilter = mount(<TodosFilter onSearch={mock1} />);
      const first = todosFilter.find('input[type="checkbox"]');
      first.node.checked = true;
      first.simulate('change', { target: { checked: true } });
      expect(mock1).toHaveBeenCalledWith({ filterByComplete: true, searchText: '' });
    });
  });
});

