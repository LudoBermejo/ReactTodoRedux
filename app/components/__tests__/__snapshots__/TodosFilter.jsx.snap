exports[`TodosFilter  renders properly 1`] = `
<div>
  <div>
    <input
      onChange={[Function]}
      placeholder="Search in todos"
      type="search" />
  </div>
  <label>
    <input
      onChange={[Function]}
      type="checkbox" />
  </label>
</div>
`;
